cmake/BuildInfo.cmake

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.14)
+
+# These values indicate "something went wrong"
+set(PRODUCT_BRANCH_ID   "0" CACHE STRING "Branch ID")
+set(PRODUCT_BUILD_ID    "0" CACHE STRING "Build ID")
+set(PRODUCT_BUILD_TYPE  "0" CACHE STRING "Build Type")
+
+if(PRODUCT_EMBED_BUILD_INFO)
+    find_package(Git QUIET)
+    if(Git_FOUND)
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} describe --always --dirty --exclude '*'
+            OUTPUT_VARIABLE GIT_REV
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            ERROR_QUIET
+        )
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} diff --exit-code
+            OUTPUT_VARIABLE GIT_DIFF_CONTENTS
+            RESULT_VARIABLE GIT_DIFF
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            ERROR_QUIET
+        )
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} describe --exact-match --tags
+            OUTPUT_VARIABLE GIT_TAG
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            ERROR_QUIET
+        )
+        string(REGEX REPLACE "[\r\n]" " " GIT_TAG "${GIT_TAG}")
+
+        if(NOT GIT_TAG)
+            set(GIT_TAG ${GIT_REV})
+        endif()
+    endif()
+endif()
+
+if(NOT GIT_REV)
+    set(GIT_REV "untracked")
+endif()
+if(NOT GIT_TAG)
+    set(GIT_TAG ${GIT_REV})
+endif()
+
+if(PRODUCT_EMBED_BUILD_TIME)
+    # Include the hash of the diff to ensure that non-commit changes trigger a build-time update.
+    string(SHA256 CURRENT_DIFF_HASH "${GIT_DIFF_CONTENTS}")
+
+    if(WIN32)
+        execute_process(
+            COMMAND powershell -Command date +%Y-%m-%d
+            OUTPUT_VARIABLE BUILD_DATE
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        execute_process(
+            COMMAND powershell -Command date +%H:%M%z
+            OUTPUT_VARIABLE BUILD_TIME
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
```

Comment:
```
Get-Date : Cannot bind parameter 'Date'. Cannot convert value "+%H:%M%z" to
type "System.DateTime". Error: "String was not recognized as a valid DateTime."
At line:1 char:6
+ date +%H:%M%z
+      ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Get-Date], ParameterBindin
   gException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.PowerSh
   ell.Commands.GetDateCommand
```

Reply:
Also, `Get-Date` apparently doesn't support `%z`, only `%Z` (which is two digits instead of four)